src/cloudsc_python/drivers/run.py

```diff
@@ -0,0 +1,119 @@
+from cloudsc4py.utils.validation import validate
+
+from config import PythonConfig, IOConfig, default_python_config, default_io_config
+from utils import print_performance, to_csv
+
+
+def core(config: PythonConfig, io_config: IOConfig) -> None:
+    hdf5_reader = HDF5Reader(config.input_file, config.data_types)
+
+    nx = config.nx or hdf5_reader.get_nlon()
+    nz = hdf5_reader.get_nlev()
+    computational_grid = ComputationalGrid(nx, 1, nz)
+
+    state = get_state(computational_grid, hdf5_reader, gt4py_config=config.gt4py_config)
+    dt = hdf5_reader.get_timestep()
+
+    yoecldp_paramaters = hdf5_reader.get_yoecldp_parameters()
+    yoethf_parameters = hdf5_reader.get_yoethf_parameters()
+    yomcst_parameters = hdf5_reader.get_yomcst_parameters()
+    yrecldp_parameters = hdf5_reader.get_yrecldp_parameters()
+
+    cloudsc = Cloudsc(
+        computational_grid,
+        yoecldp_paramaters,
+        yoethf_parameters,
+        yomcst_parameters,
+        yrecldp_parameters,
+        enable_checks=config.sympl_enable_checks,
+        gt4py_config=config.gt4py_config,
+    )
+    tends, diags = cloudsc(state, dt)
+
+    runtimes = []
+    for i in range(config.num_runs):
+        with timing(f"run_{i}") as timer:
+            cloudsc(state, dt, out_tendencies=tends, out_diagnostics=diags)
+        runtimes.append(timer.get_time(f"run_{i}", units="ms"))
+
+    runtime_mean, runtime_stddev = print_performance(runtimes)
+
+    if io_config.output_file is not None:
+        to_csv(
+            io_config.output_file,
+            io_config.host_name,
+            config.gt4py_config.backend,
+            config.nx,
+            config.num_runs,
+            runtime_mean,
+            runtime_stddev,
+        )
+
+    if config.enable_validation:
+        hdf5_reader_ref = HDF5Reader(config.reference_file, config.data_types)
+        tends_ref = get_reference_tendencies(
+            computational_grid, hdf5_reader_ref, gt4py_config=config.gt4py_config
+        )
+        diags_ref = get_reference_diagnostics(
+            computational_grid, hdf5_reader_ref, gt4py_config=config.gt4py_config
+        )
+
+        tends_fail = validate(tends, tends_ref)
+        if len(tends_fail) == 0:
+            print("Results: All tendencies have been successfully validated. HOORAY!")
+        else:
+            print(
+                f"Results: Validation failed for {len(tends_fail)}/{len(tends_ref) - 1} "
+                f"tendencies: {', '.join(tends_fail)}."
+            )
+
+        diags_fail = validate(diags, diags_ref)
+        if len(diags_fail) == 0:
+            print("Results: All diagnostics have been successfully validated. HOORAY!")
+        else:
+            print(
+                f"Results: Validation failed for {len(diags_fail)}/{len(diags_ref) - 1} "
+                f"diagnostics: {', '.join(diags_fail)}."
+            )
+
+
+@click.command()
+@click.option("--backend", type=str, default=None)
```

Comment:
Added a description for each option, specifying admissible, recommended and default values. Regarding multi-threading, for GT4Py this can be controlled using the env variable `OMP_NUM_THREADS`, as done in `drivers/run_batch.sh`.